File: src/snippets/rate-limiting/reference/sveltekit/ByUserId.js
import { env } from "$env/dynamic/private";
import arcjet, { fixedWindow } from "@arcjet/sveltekit";
import { error, json } from "@sveltejs/kit";

const aj = arcjet({
  key: env.ARCJET_KEY,
  rules: [
    fixedWindow({
      mode: "LIVE",
      characteristics: ["userId"],
      window: "1h",
      max: 60,
    }),
  ],
});

export async function GET(event) {
  // Pass userId as a string to identify the user. This could also be a number
  // or boolean value.
  const decision = await aj.protect(event, { userId: "user123" });

  if (decision.isDenied()) {
    return error(429, { message: "Too many requests" });
  }

  return json({ message: "Hello world" });
}

File: src/snippets/rate-limiting/reference/python-fastapi/ByUserId.py
import os

from arcjet import Mode, arcjet, fixed_window
from fastapi import FastAPI, Request
from fastapi.responses import JSONResponse

app = FastAPI()

aj = arcjet(
    key=os.environ["ARCJET_KEY"],  # Get your site key from https://app.arcjet.com
    rules=[
        fixed_window(
            mode=Mode.LIVE,
            characteristics=["userId"],
            window=3600,  # 1 hour in seconds
            max=60,
        ),
    ],
)


@app.get("/")
async def index(request: Request):
    # Pass userId as a string to identify the user. This could also be a number
    # or boolean value
    decision = await aj.protect(request, characteristics={"userId": "user123"})
    print("Arcjet decision", decision)

    if decision.is_denied():
        return JSONResponse({"error": "Too Many Requests"}, status_code=429)

    return {"message": "Hello world"}
